### PR TITLE
Fix backbone freezing with icevision models

### LINKED ISSUE #1080: Trainer.finetune broken for Keypoint detection module
## 🐛 Bug

Finetuning the `KeypointDetector` with freeze strategy does not freeze the parameters.

### To Reproduce

Follow the [Keypoint detection example]( https://lightning-flash.readthedocs.io/en/latest/reference/keypoint_detection.html)

#### Code sample

```python
import flash
print(flash.__version__)
from flash.core.utilities.imports import example_requires
from flash.image import KeypointDetectionData, KeypointDetector

example_requires("image")

import icedata  # noqa: E402

data_dir = icedata.biwi.load_data()

datamodule = KeypointDetectionData.from_icedata(
    train_folder=data_dir,
    val_split=0.1,
    batch_size=4,
    parser=icedata.biwi.parser,
)

# 2. Build the task
model = KeypointDetector(
    head="keypoint_rcnn",
    backbone="resnet18_fpn",
    num_keypoints=1,
    num_classes=datamodule.num_classes,
)

# 3. Create the trainer and finetune the model
trainer = flash.Trainer(max_epochs=100, gpus=1)
trainer.finetune(model, datamodule=datamodule, strategy="freeze")

trainable = 0
for name, param in model.adapter.model.named_parameters(): # or try from torchsummary import summary?
    if param.requires_grad:
        o = (name, np.product(param.shape), param.requires_grad)
        if o[2]:
            trainable += o[1]
        print(*o)
print(trainable)
    
```

### Expected behavior

The number of trainable parameters should be smaller than 45.8M. Also, `trainer.fit` takes about as much time as `trainer.finetune`:

```
  | Name          | Type                              | Params
--------------------------------------------------------------------
0 | train_metrics | ModuleDict                        | 0     
1 | val_metrics   | ModuleDict                        | 0     
2 | test_metrics  | ModuleDict                        | 0     
3 | adapter       | IceVisionKeypointDetectionAdapter | 46.0 M
--------------------------------------------------------------------
45.8 M    Trainable params
156 K     Non-trainable params
46.0 M    Total params
183.924   Total estimated model params size (MB)
```

### Environment

```
efficientnet-pytorch          0.6.3
icevision                     0.11.0
lightning-bolts               0.4.0
lightning-flash               0.7.0.dev0
pytorch-lightning             1.5.6
segmentation-models-pytorch   0.2.1
torch                         1.10.0+cu111
torchaudio                    0.10.0+cu111
torchmetrics                  0.6.2
torchsummary                  1.5.1
torchtext                     0.11.0
torchvision                   0.11.1+cu111
```

### PR DESCRIPTION
**Pull request by ethanwharris**
## What does this PR do?

Fixes #1080 

## Before submitting
- [x] Was this discussed/approved via a Github issue? (no need for typos and docs improvements)
- [x] Did you read the [contributor guideline](https://github.com/PyTorchLightning/lightning-flash/tree/master/.github/CONTRIBUTING.md), Pull Request section?
- [x] Did you make sure your PR does only one thing, instead of bundling different changes together?
- [ ] Did you make sure to update the documentation with your changes?
- [ ] Did you write any new necessary tests? [not needed for typos/docs]
- [ ] Did you verify new and existing tests pass locally with your changes?
- [ ] If you made a notable change (that affects users), did you update the [CHANGELOG](https://github.com/PyTorchLightning/lightning-flash/blob/master/CHANGELOG.md)?

## PR review
 - [x] Is this pull request ready for review? (if not, please submit in draft mode)

Anyone in the community is free to review the PR once the tests have passed.
If we didn't discuss your PR in Github issues there's a high chance it will not be merged.

## Did you have fun?
Make sure you had fun coding 🙃

### CODE REVIEW

File: flash/image/detection/model.py [hunk 1]
```diff
@@ -49,7 +49,7 @@ def __init__(
         pretrained: bool = True,
         optimizer: OPTIMIZER_TYPE = "Adam",
         lr_scheduler: LR_SCHEDULER_TYPE = None,
-        learning_rate: float = 5e-3,
+        learning_rate: float = 1e-2,
```
Comment (ligaz) [inline flash/image/detection/model.py:52]: @ethanwharris Can you elaborate why the default learning rate has been changed here? Thanks.

Comment (ethanwharris) [reply flash/image/detection/model.py:52]: Hey, @ligaz sorry just an accidental change 😃 Woud you prefer the old default? Or something else? Please feel free to open a PR to revert / update it

Comment (ligaz) [reply flash/image/detection/model.py:52]: It was a general question what's the reason behind this change.

We are defaulting the optimizer to `Adam` so my suggestion is to use the default learning rate for it. From the PyTorch docs [here](https://pytorch.org/docs/stable/generated/torch.optim.Adam.html):
```
learning rate (default: 1e-3)
```

Comment (ethanwharris) [reply flash/image/detection/model.py:52]: I think that makes sense! I wonder if there's a way we could just not provide the LR to the optimizer constructor if it is `None`? That way if you don't override the LR you would always just get the default for your chosen optimizer. What do you think?

Comment (ligaz) [reply flash/image/detection/model.py:52]: I was thinking about the same. It turns out that it defaults to the code in the core Task class which uses `5e-5` as a default value. This is the relevant [code](https://github.com/PyTorchLightning/lightning-flash/blob/master/flash/core/model.py#L308).

I'm personally not sure how we should proceed 😄

Comment (ethanwharris) [reply flash/image/detection/model.py:52]: It looks like we could jsut default the `learning_rate` to `None`, and then adjust the logic here: https://github.com/PyTorchLightning/lightning-flash/blob/4c624823d7992aa454b9f064d2714cae9b4a8893/flash/core/model.py#L463

If learning rate is None then the `optimizer_kwargs` should just be an empty dict. That would do it I think. Let me know if you'd like to try it otherwise I can take it :smiley:

**Comment by ligaz** [comment]
Feel free to take it. Thanks for your support 🙏

File: flash/image/detection/model.py (hunk 1 above)
Comment (ligaz) [reply flash/image/detection/model.py:52]: Feel free to take it. Thanks for your support 🙏
